wizard-frontend/src/components/steps/TargetTab.tsx

```diff
@@ -0,0 +1,205 @@
+import { useState, useContext } from "react";
+import { Server, AlertCircle, ChevronDown } from "lucide-react";
+import { Button } from "../ui/button";
+import { Card, CardContent, CardHeader, CardTitle } from "../ui/card";
+import { Badge } from "../ui/badge";
+import { Input } from "../ui/input";
+import { Label } from "../ui/label";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "../ui/select";
+import { Popover, PopoverContent, PopoverTrigger } from "../ui/popover";
+import { TabsContent } from "../ui/tabs";
+import {
+  readCurrentTargetDetails,
+  updateConfigPorts,
+  updateConfigTarget,
+} from "../JsonUtils";
+import { ConfigDataContext } from "../UserDataContext";
+import { useQuery } from "@tanstack/react-query";
+import ALL_API_ROUTES from "../../lib/routes";
+
+// interface ClusterDetails {
+//   namespaces: string[];
+//   target_types: string[];
+// }
+
+interface Target {
+  target_path: string;
+  target_namespace: string;
+  detected_ports: number[];
+}
+
+const TargetTab = ({ setTargetPorts }: { setTargetPorts : (ports: number[]) => void }) => {
+  const { config, setConfig } = useContext(ConfigDataContext)!;
+  const [namespace, setNamespace] = useState<string>("default");
+  const [targetType, setTargetType] = useState<string>("");
+  const [targetSearchText, setTargetSearchText] = useState<string>("");
+
+  // refresh cluster details every 30 seconds
+  const clusterDetailsQuery = useQuery({
+    staleTime: 30 * 1000,
+    queryKey: ["clusterDetails"],
+    queryFn: () =>
+      fetch(window.location.origin + ALL_API_ROUTES.clusterDetails).then(
+        async (res) =>
+          res.ok ? await res.json() : { namespaces: [], target_types: [] }
+      ),
+  });
+
+  if (clusterDetailsQuery.error) console.log(clusterDetailsQuery.error);
+
+  const availableNamespaces: string[] =
+    clusterDetailsQuery.isLoading || clusterDetailsQuery.error
+      ? []
+      : clusterDetailsQuery.data.namespaces;
+  const availableTargetTypes: string[] =
+    clusterDetailsQuery.isLoading || clusterDetailsQuery.error
+      ? []
+      : clusterDetailsQuery.data.target_types;
+
+  const targetsQuery = useQuery({
```

Comment:
You know what would be really cool?

Target selection done by steps (not necessarily new tabs or anything like that). So you select a `Namespace`, and we only show you `Target Type`s that are available in that `Namespace` (like, why am I able to select `cronjob` if there aren't any in the `Namespace` I selected?).

Maybe it would be too much for this version? Idk how soon we want to release this.